subsys/bluetooth/mesh/cfg_srv.c

```diff
@@ -2039,10 +2039,17 @@ static int mod_app_get(struct bt_mesh_model *model,
 	if (mod) {
 		int i;
 
-		for (i = 0; i < mod->keys_cnt; i++) {
-			if (mod->keys[i] != BT_MESH_KEY_UNUSED) {
-				net_buf_simple_add_le16(&msg, mod->keys[i]);
+		for (i = 0; i < mod->keys_cnt - 1; i += 2) {
+			if (mod->keys[i] == BT_MESH_KEY_UNUSED &&
+			    mod->keys[i + 1] == BT_MESH_KEY_UNUSED) {
+				break;
 			}
+
+			key_idx_pack(&msg, mod->keys[i], mod->keys[i + 1]);
+		}
+
+		if (i < mod->keys_cnt && mod->keys[i] != BT_MESH_KEY_UNUSED) {
```

Comment:
Probably it is worth to modify `key_idx_pack` to be able to handle `BT_MESH_KEY_UNUSED` instead of adding unique handler for the last key.

Reply:
I'm restructuring a bit now. Lets see what you think after I push it.